### PR TITLE
Post merge cleanup 36229 ember buffer encode

### PR DESCRIPTION
**Pull request by andy31415**
Fixes post review comments from @bzbarsky-apple on #36229

### CODE REVIEW

File: src/app/codegen-data-model-provider/EmberAttributeDataBuffer.cpp [hunk 1]
```diff
@@ -384,29 +384,22 @@ CHIP_ERROR EmberAttributeDataBuffer::EncodeInteger(chip::TLV::TLVWriter & writer
...
 
-    bool isSigned = (mAttributeType == ZCL_INT8S_ATTRIBUTE_TYPE) //
-        || (mAttributeType == ZCL_INT16S_ATTRIBUTE_TYPE)         //
-        || (mAttributeType == ZCL_INT24S_ATTRIBUTE_TYPE)         //
-        || (mAttributeType == ZCL_INT32S_ATTRIBUTE_TYPE)         //
-        || (mAttributeType == ZCL_INT40S_ATTRIBUTE_TYPE)         //
-        || (mAttributeType == ZCL_INT48S_ATTRIBUTE_TYPE)         //
-        || (mAttributeType == ZCL_INT56S_ATTRIBUTE_TYPE)         //
-        || (mAttributeType == ZCL_INT64S_ATTRIBUTE_TYPE);
+    const bool isSigned = chip::app::IsSignedAttributeType(mAttributeType);
```
Comment (bzbarsky-apple) [inline src/app/codegen-data-model-provider/EmberAttributeDataBuffer.cpp:387]:
```suggestion
    const bool isSigned = IsSignedAttributeType(mAttributeType);
```

File: src/app/codegen-data-model-provider/EmberAttributeDataBuffer.cpp [hunk 2]
```diff
@@ -445,35 +438,33 @@ CHIP_ERROR EmberAttributeDataBuffer::EncodeInteger(chip::TLV::TLVWriter & writer
...
+    case ZCL_INT8U_ATTRIBUTE_TYPE:  // Unsigned 8-bit integer
     case ZCL_INT16U_ATTRIBUTE_TYPE: // Unsigned 16-bit integer
-        return writer.Put(tag, static_cast<uint16_t>(value.uint_value));
     case ZCL_INT24U_ATTRIBUTE_TYPE: // Unsigned 24-bit integer
     case ZCL_INT32U_ATTRIBUTE_TYPE: // Unsigned 32-bit integer
-        return writer.Put(tag, static_cast<uint32_t>(value.uint_value));
     case ZCL_INT40U_ATTRIBUTE_TYPE: // Unsigned 40-bit integer
     case ZCL_INT48U_ATTRIBUTE_TYPE: // Unsigned 48-bit integer
     case ZCL_INT56U_ATTRIBUTE_TYPE: // Signed 56-bit integer
     case ZCL_INT64U_ATTRIBUTE_TYPE: // Signed 64-bit integer
```
Comment (bzbarsky-apple) [inline src/app/codegen-data-model-provider/EmberAttributeDataBuffer.cpp:460]:
```suggestion
    case ZCL_INT56U_ATTRIBUTE_TYPE: // Unsigned 56-bit integer
    case ZCL_INT64U_ATTRIBUTE_TYPE: // Unsigned 64-bit integer
```

**Review by bzbarsky-apple — COMMENTED** [review]
https://github.com/project-chip/connectedhomeip/pull/36229#discussion_r1817452386 does not seem to be addressed.

File: src/app/codegen-data-model-provider/EmberAttributeDataBuffer.cpp [hunk 3]
```diff
@@ -445,35 +438,33 @@ CHIP_ERROR EmberAttributeDataBuffer::EncodeInteger(chip::TLV::TLVWriter & writer
...
     }
 
-    if (mIsNullable && (value.uint_value == nullValue))
+    // We place the null value as either int_value or uint_value into a union that is
+    // bit-formatted as both int64 and uint64. When we define the nullValue,
+    // it is bitcast into u64 hence this comparison. This is ugly, however this
+    // code prioritizes code size over readability here.
+    if (mIsNullable && (value.uint_value == nullValueAsU64))
     {
         // MaxValue is used for NULL setting
```
[the remark this answers is not in the document]
Comment (andy31415) [reply src/app/codegen-data-model-provider/EmberAttributeDataBuffer.cpp:447]: I added a bit comment above and forgot to remove this... will remove

File: src/app/codegen-data-model-provider/EmberAttributeDataBuffer.cpp [hunk 4]
```diff
@@ -445,35 +438,33 @@ CHIP_ERROR EmberAttributeDataBuffer::EncodeInteger(chip::TLV::TLVWriter & writer
...
     case ZCL_INT64U_ATTRIBUTE_TYPE: // Signed 64-bit integer
-        return writer.Put(tag, static_cast<uint64_t>(value.uint_value));
-    case ZCL_INT8S_ATTRIBUTE_TYPE: // Signed 8-bit integer
-        return writer.Put(tag, static_cast<int8_t>(value.int_value));
+        return writer.Put(tag, value.uint_value);
+    case ZCL_INT8S_ATTRIBUTE_TYPE:  // Signed 8-bit integer
     case ZCL_INT16S_ATTRIBUTE_TYPE: // Signed 16-bit integer
-        return writer.Put(tag, static_cast<int16_t>(value.int_value));
     case ZCL_INT24S_ATTRIBUTE_TYPE: // Signed 24-bit integer
     case ZCL_INT32S_ATTRIBUTE_TYPE: // Signed 32-bit integer
```
[the remark this answers is not in the document]
Comment (andy31415) [reply src/app/codegen-data-model-provider/EmberAttributeDataBuffer.cpp]: updated.

**Comment by andy31415** [comment]
> [#36229 (comment)](https://github.com/project-chip/connectedhomeip/pull/36229#discussion_r1817452386) does not seem to be addressed.

Done. Did not see header comments. The comment was a copy and paste and it turned out handling both signed and unsigned into a single method was smaller code.

**Review by bzbarsky-apple — APPROVED** [review]

**Review by cecille — APPROVED** [review]

**Review by jmartinez-silabs — APPROVED** [review]

**Review by tehampson — APPROVED** [review]

**Pull request merged**